ion/integration_test.go

```diff
@@ -0,0 +1,807 @@
+	if !reflect.DeepEqual(txtWriter, txtWriter2) {
+		t.Errorf("Round trip test failed on: " + fp)
+	}
+}
+
+// Create a TextWriter from data parameter. Return the writer and string builder containing writer's contents
+func makeTextWriter(t *testing.T, data string) (Writer, strings.Builder) {
+	reader := NewReader(strings.NewReader(data))
+	str := strings.Builder{}
+	txtWriter := NewTextWriter(&str)
+	writeToWriterFromReader(t, reader, txtWriter)
+	txtWriter.Finish()
+	return txtWriter, str
+}
+
+// Create a BinaryWriter from data parameter. Return the writer and buffer containing writer's contents
+func makeBinaryWiter(t *testing.T, data []byte) (Writer, bytes.Buffer) {
+	reader := NewReader(bytes.NewReader(data))
+	buf2 := bytes.Buffer{}
+	binWriter2 := NewBinaryWriter(&buf2)
+	writeToWriterFromReader(t, reader, binWriter2)
+	binWriter2.Finish()
+	return binWriter2, buf2
+}
+
+// Execute loading malformed Ion values into a Reader and validate the Reader.
+func testLoadBad(t *testing.T, fp string) {
+	file, er := os.Open(fp)
+	if er != nil {
+		t.Fatal(er)
+	}
+	defer file.Close()
+
+	r := NewReader(file)
+
+	err := testInvalidReader(t, r)
+
+	if r.Err() == nil && err == nil {
+		t.Fatal("Should have failed loading \"" + fp + "\".")
+	} else {
+		t.Log("expectedly failed loading " + r.Err().Error())
+	}
+}
+
+// Traverse the reader and check if it is an invalid reader, containing malformed Ion values.
+func testInvalidReader(t *testing.T, r Reader) error {
+	for r.Next() {
+		switch r.Type() {
+		case StructType, ListType, SexpType:
+			r.StepIn()
+			testInvalidReader(t, r)
+			r.StepOut()
+		}
+	}
+	if r.Err() != nil {
+		return r.Err()
+	}
+
+	return nil
+}
+
+// Execute equivalency and non-equivalency tests, where true for eq means
+// equivalency and false denotes non-equivalency test.
+func testEquivalency(t *testing.T, fp string, eq bool) {
+	file, er := os.Open(fp)
+	if er != nil {
+		t.Fatal(er)
+	}
+	defer file.Close()
+
+	r := NewReader(file)
+	for r.Next() {
+		embDoc := embeddedDoc(r.Annotations())
+		switch r.Type() {
+		case StructType, ListType, SexpType:
+			var values []item
+			r.StepIn()
+			if embDoc {
+				values = handleEmbeddedDoc(t, r)
+			} else {
+				for r.Next() {
+					values = append(values, readCurrentValue(t, r))
+				}
+			}
+			equivalencyAssertion(t, values, eq)
+			r.StepOut()
+		}
+	}
+	if r.Err() != nil {
+		t.Error()
+	}
+}
+
+// Ion values in top-level sequence annotated with "embedded_documents" parse as Ion string and each
+// of the string values in turn, parse as a document and put in an item struct (defined in this file).
+func handleEmbeddedDoc(t *testing.T, r Reader) []item {
+	var values []item
+	for r.Next() {
+		str, err := r.StringValue()
+		if err != nil {
+			t.Error("Must be string value.")
+		}
+		newReader := NewReaderStr(str)
+		for newReader.Next() {
+			values = append(values, readCurrentValue(t, newReader))
+		}
+	}
+	return values
+}
+
+func embeddedDoc(an []string) bool {
```

Comment:
Per the Go style guidelines, I suggest renaming this "isEmbeddedDoc" as it returns a `bool`.